client/ayon_core/plugins/publish/extract_otio_review.py

```diff
@@ -99,48 +100,68 @@ def process(self, instance):
-            if height != self.to_height:
-                self.to_height = height
-
+            res_data = {"width": self.to_width, "height": self.to_height}
+            for key in res_data:
+                for meta_prefix in ("ayon.source.", "openpype.source."):
+                    meta_key = f"{meta_prefix}.{key}"
+                    value = media_metadata.get(meta_key)
+                    if value is not None:
+                        res_data[key] = value
+                        break
+
+            self.to_width, self.to_height = res_data["width"], res_data["height"]
             self.log.debug("> self.to_width x self.to_height: {} x {}".format(
                 self.to_width, self.to_height
             ))
 
-            # get frame range values
+            # Clip: compute process range from available media range.
             src_range = r_otio_cl.source_range
-            start = src_range.start_time.value
-            duration = src_range.duration.value
-            available_range = None
-            self.actual_fps = src_range.duration.rate
-
-            # add available range only if not gap
             if isinstance(r_otio_cl, otio.schema.Clip):
                 available_range = r_otio_cl.available_range()
+                processing_range = None
                 self.actual_fps = available_range.duration.rate
+                start = src_range.start_time.rescaled_to(self.actual_fps)
+                duration = src_range.duration.rescaled_to(self.actual_fps)
+
+                # Temporary.
+                # Some AYON custom OTIO exporter were implemented with relative
+                # source range for image sequence. Following code maintain
+                # backward-compatibility by adjusting available range
+                # while we are updating those.
+                media_ref = r_otio_cl.media_reference
+                if (
+                    is_clip_from_media_sequence(r_otio_cl)
+                    and available_range.start_time.to_frames() == media_ref.start_frame
+                    and src_range.start_time.to_frames() < media_ref.start_frame
+                ):
+                    available_range = otio.opentime.TimeRange(
+                        otio.opentime.RationalTime(0, rate=self.actual_fps),
+                        available_range.duration,
+                    )
+
+            # Gap: no media, generate range based on source range
+            else:
+                available_range = processing_range = None
+                self.actual_fps = src_range.duration.rate
+                start = src_range.start_time
+                duration = src_range.duration
+
+            # Create handle offsets.
+            handle_start = otio.opentime.RationalTime(
+                handle_start,
+                rate=self.actual_fps,
+            )
+            handle_end = otio.opentime.RationalTime(
+                handle_end,
+                rate=self.actual_fps,
+            )
```

Comment:
This seems to be breaking the backward compatibility in my Hiero testing scene:

It looks like the **handle_start** and **handle_end** (originally intager) is overriden here into `opentime.RT` type.

```
Traceback (most recent call last):
  File "C:\Users\jakub\AppData\Local\Ynput\AYON\dependency_packages\ayon_2408091537_windows.zip\dependencies\pyblish\plugin.py", line 528, in __explicit_process
    runner(*args)
  File "C:\CODE\__YNPUT\ayon-core\client\ayon_core\plugins\publish\extract_otio_review.py", line 158, in process
TypeError: float() argument must be a string or a real number, not 'opentimelineio._opentime.RationalTime'
```

Reply:
Thanks a lot for the double check, I completely forgot about retesting the multi-review-clips case. Totally my bad, apologies for this.

I've fixed the problem and added 2 additional unit tests to enforce those.
Also confirmed that it worked OK on Hiero (Linux) and Resolve (Windows).